app/models/registration/campaign.rb

```diff
@@ -26,5 +26,17 @@ class Campaign < ApplicationRecord
                     completed: 3 }
 
     validates :title, :registration_deadline, :allocation_mode, :status, presence: true
+
+    def evaluate_policies_for(user, phase: :registration)
+      Registration::PolicyEngine.new(self).eligible?(user, phase: phase)
+    end
+
+    def policies_satisfied?(user, phase: :registration)
+      evaluate_policies_for(user, phase: phase).pass
+    end
+
+    def user_registered?(user)
```

Comment:
I propose to rename this to something containing "confirmed", e.g. `user_registration_confirmed?` for more clarity.